src/Servers/Kestrel/Core/src/Internal/Infrastructure/Heartbeat.cs

```diff
@@ -62,17 +62,20 @@ internal void OnHeartbeat()
         }
         catch (Exception ex)
         {
-            _trace.LogError(0, ex, $"{nameof(Heartbeat)}.{nameof(OnHeartbeat)}");
+            if (!_stopped)
+            {
+                _trace.LogError(0, ex, $"{nameof(Heartbeat)}.{nameof(OnHeartbeat)}");
+            }
         }
     }
 
     private void TimerLoop()
     {
+        Thread.Sleep(_interval);
```

Comment:
I added an initial `Thread.Sleep` before the timer loop to avoid a potential breaking change. The previous code guaranteed a `Thread.Sleep` call before the first invocation of `OnHeatbeat`. I wanted to avoid the possibility of any latent issues of a heartbeat handler failing if called too soon. This change maintains this behavior.